### PR TITLE
Added www-authenticate: bearer if the access token is invalid

### LINKED ISSUE #32882: API : External Auth Server : Returns (not) a header `WWW-Authenticate : Bearer` if the access token is invalid
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When using an external Auth Server, the API doesn't return a header `WWW-Authenticate : Bearer` if the access token is invalid

### Expected behavior

When using an external Auth Server, the API returns a header `WWW-Authenticate : Bearer` if the access token is invalid

The CI must be green : `TEST_PATH="functional/API/01*/02*/02*" npm run test:specific`

### Steps to reproduce

1. Launch a keycloak : `docker run -p 8003:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:19.0.3 start-dev`
2. Install the module KeyCloak Connector Demo
3. Configure the realm endpoint : http://127.0.0.1:8003/realms/master
4. Request the endpoint `http://domain.tld/admin-dev/new-api/hook-status/1` with header `Authorization: 'Bearer INVALIDTOKEN'`
5. The HTTP Code received is a 401
6. The header  `WWW-Authenticate : Bearer` is not returned

#### Acceptance Test
Unskip the test : 
https://github.com/PrestaShop/PrestaShop/blob/d672ad4ccc467722d13746e25fffc95a86e5e3b3/tests/UI/campaigns/functional/API/01_clientCredentialGrantFlow/02_externalAuthServer/02_resourceEndpoint.ts#L136-L137

### PrestaShop version(s) where the bug happened

8.1.x

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

keycloak_connector_demo v1.0.2

### Your company or customer's name goes here (if applicable).

@PrestaShopCorp

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x 
| Description?      | Returns a header WWW-Authenticate : Bearer if the access token is invalid
| Type?             | bug fix 
| Category?         | WS 
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | https://github.com/PrestaShop/PrestaShop/issues/32882
| Fixed ticket?     | Fixes #32882
| Related PRs       | ~
| Sponsor company   | PrestaShop SA

### CODE REVIEW

**Comment by nicosomb** [comment]
Tests https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/5243891797

File: tests/UI/campaigns/functional/API/01_clientCredentialGrantFlow/02_externalAuthServer/02_resourceEndpoint.ts [hunk 1]
```diff
@@ -134,7 +134,7 @@ describe('API : External Auth Server - Resource Endpoint', async () => {
     });
 
     // @todo : https://github.com/PrestaShop/PrestaShop/issues/32768
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/API/01_clientCredentialGrantFlow/02_externalAuthServer/02_resourceEndpoint.ts]:
```suggestion
```

**Review by mflasquin — DISMISSED** [review]

**Review by Progi1984 — CHANGES_REQUESTED** [review]
The CI is :red_circle:

**Review by jolelievre — DISMISSED** [review]
Thanks @nicosomb

**Review by M0rgan01 — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Comment by nicosomb** [comment]
Tests https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/5280351396

**Comment by nicosomb** [comment]
Test https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/5280636880

**Comment by djoelleuch** [comment]
Hello @nicosomb , 
some checks were not successful , could you please check it  .

**Comment by Progi1984** [comment]
UI Test : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/5517067057

**Comment by nicosomb** [comment]
UI tests   https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/5643118513

**Review by Progi1984 — APPROVED** [review]
Approved for UI tests (Unskip the test as defined in Acceptance Test and UI tests are :green_circle:)

Ping @PrestaShop/ui-tests-maintainers

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by M0rgan01 — APPROVED** [review]

**Review by mflasquin — APPROVED** [review]

**Review by djoelleuch — APPROVED** [review]
Hello @nicosomb , 

I have tested your PR and it's well fixed !

[Untitled_ Jul 27, 2023 1_57 PM.webm](https://github.com/PrestaShop/PrestaShop/assets/115152460/0b5af7a3-ea7c-4ed2-8765-d8b0a5a2b302)


automated tests : OK 
It's QA approved :heavy_check_mark:
Thank you :rocket:

**Comment by Progi1984** [comment]
Congrats @nicosomb : Never give up ;)

**Comment by matthieu-rolland** [comment]
thank you @nicosomb and @djoelleuch

**Pull request merged**